### DISCUSSION TITLE
Rework Slack notification channel configuration

### DISCUSSION THREAD

**Discussion by yuri-tceretian** [grafana/grafana#40829] [Alerting]
The current version of the form for configuring the Slack notification channel causes confusion for people who sees it the first time or probably even those who do not use it often. 
<details><summary> Form </summary>

![image](https://user-images.githubusercontent.com/25988953/138508647-4b184a64-f0b8-42ed-a1ce-5225537c65ba.png)

</details>

The form contains so many fields and their relationship is not obvious. The form lets users configure the Slack channel to send notifications using at least two different ways: Incoming Webhooks and chat API. In the former case, the user needs to just enter the URL at the very bottom, whereas in the latter case token + recipient. Other fields seem to be optional. 

When I had to configure the notification channel the first time it took me a lot of time reading through docs and trying to understand what the fields in that form are for and how I can configure it. 

Also, the complexity of the form makes some users to incorrectly configure the notifications (the migration to ngalert revealed many (10-20) users that misconfigured it (Grafana let them because of a bug in API that lets users save invalid changes that would not work if they click "Test").

I think this form can be radically simplified. 
1. Separate use-cases: have different areas for configuring Incoming Webhook and Chat API. 
2. Add help links to documentation of how to obtain the info required for configuring for every use-case.
3. Make optional fields more obvious. Probably hide them under collapsible panel.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
